tools/asset-sync/assets.Tests.ps1

```diff
@@ -0,0 +1,430 @@
+
+      Initialize-AssetsRepo -Config $config
+
+      $parsedResult = Is-AssetsRepo-Initialized -Config $config
+      $parsedResult | Should -Be $true
+    }
+
+    It "Should initialize language repo with a new assets.json at sdk/ if necessary" {
+      # TODO, do we even need this?
+    }
+  }
+
+  # each of the tests in this context is resolving one of the known cases when pushing changes to an auto commit branch
+  # In order:
+  #    - Auto Branch Doesn't Exist Yet, Go Off Main, push to new branch
+  #    - Auto Branch Exists, we're on the latest commit, push new commit to branch
+  #    - Auto Branch Exists, we're on a commit from the past, push new commit to branch
+  Context "Push-AssetsRepo-Update" {
+    It "Should push a new branch/commit to a non-existent target branch." {
+      $sourceBranch = "scenario_new_push"
+      $recordingJson = [PSCustomObject]@{
+        AssetsRepo = "Azure/azure-sdk-assets-integration"
+        AssetsRepoPrefixPath = "python/recordings/"
+        AssetsRepoId = ""
+        AssetsRepoBranch = "$sourceBranch"
+        SHA = "786b4f3d380d9c36c91f5f146ce4a7661ffee3b9"
+      }
+
+      $files = @(
+        "sdk/tables/azure-data-tables/tests/recordings/test_retry.pyTestStorageRetrytest_retry_on_timeout.json",
+        "sdk/tables/azure-data-tables/tests/recordings/test_retry.pyTestStorageRetrytest_retry_on_server_error.json",
+        "sdk/tables/azure-data-tables/assets.json"
+      )
+
+      $testFolder = Describe-TestFolder -AssetsJsonContent $recordingJson -Files $files -IntegrationBranch $recordingJson.AssetsRepoBranch
+      $config = Resolve-AssetsJson (Join-Path $testFolder "sdk" "tables" "azure-data-tables")
+
+      Initialize-AssetsRepo -Config $config
+      $assetRepoFolder = Resolve-AssetRepo-Location -Config $config
+      
+      foreach($file in $files){
+        $sourcePath = Join-Path $testFolder $file 
+        $targetPath = Join-Path $assetRepoFolder $config.AssetsRepoPrefixPath $file
+
+        $targetFolder = Split-Path $targetPath
+        if(-not (Test-Path $targetFolder)){
+          mkdir -p $targetFolder
+        }
+
+        Copy-Item  -Force -Path $sourcePath -Destination $targetPath
+      }
+      $config.AssetsRepoBranch | Should -not -Be $sourceBranch
+
+      Push-AssetsRepo-Update -Config $Config
+
+      try {
+        Push-Location $assetRepoFolder
+        $repoBranchSHA = git rev-parse origin/$($config.AssetsRepoBranch) --quiet
+      }
+      finally {
+        Pop-Location
+      }
+
+      # re-parse from the on-disk json
+      $configReparsed = Resolve-AssetsJson (Join-Path $testFolder "sdk" "tables" "azure-data-tables")
+
+      # re-parsed config from disk should
+      #  -> match the updated sha
+      #  -> match the SHA we get back from the repo
+      $repoBranchSHA | Should -Be $config.SHA
+      $configReparsed.SHA | Should -Be $config.SHA
+    }
+    
+    It "Should push a clean new commmit to the target branch." {
```

Comment:
It might be worth separating these integration type tests into their own context or file so it's easy to just run the unit test suite if needed?

Reply:
Yes! I will use tags and our CI will run tests tagged with `Unit` versus `Integration`.